File: _assets/scripts/normalizers/texture_list_normalize.lua
return function (data)
    local output = python.builtins.dict()
    local output_attr = python.as_attrgetter(output)
    local data_attr = python.as_attrgetter(data)
    for index1, key_value1 in python.enumerate(data_attr.items()) do
        local resource_pack = key_value1[0]
        local textures = key_value1[1]
        for index, texture in python.enumerate(textures) do
            if not output_attr.__contains__(texture) then
                output[texture] = python.builtins.list()
            end
            python.as_attrgetter(output[texture]).append(resource_pack)
        end
    end
    return output
end

File: _assets/scripts/normalizers/texture_list_normalize.py
from collections import defaultdict

import Utilities.File as File

__all__ = ["texture_list_normalize"]

def texture_list_normalize(data:dict[str,File.File[list[str]]]) -> File.FakeFile[dict[str,list[str]]]:
    output:defaultdict[str,list[str]] = defaultdict(lambda: [])
    file_hashes:list[int] = []
    for resource_pack, textures_file in data.items():
        textures = textures_file.data
        file_hashes.append(hash(textures_file))
        for texture in textures:
            output[texture].append(resource_pack)
    return File.FakeFile("combined_texture_list_file", dict(output), hash(tuple(file_hashes)))
